executor/builder.go

```diff
@@ -1003,33 +1003,59 @@ func (b *executorBuilder) buildHashJoin(v *plannercore.PhysicalHashJoin) Executo
 		joinType:          v.JoinType,
 		isOuterJoin:       v.JoinType.IsOuterJoin(),
 		buildSideEstCount: v.Children()[v.InnerChildIdx].StatsCount(),
+		useOuterToBuild:   v.UseOuterToBuild,
 	}
-
 	defaultValues := v.DefaultValues
 	lhsTypes, rhsTypes := retTypes(leftExec), retTypes(rightExec)
-	if v.InnerChildIdx == 0 {
-		if len(v.LeftConditions) > 0 {
-			b.err = errors.Annotate(ErrBuildExecutor, "join's inner condition should be empty")
-			return nil
+	if v.UseOuterToBuild {
```

Comment:
We can minimize the code.
```go
e.buildSideExec = leftExec
e.probeSideExec = rightExec
e.buildKeys = v.LeftJoinKeys
e.probeKeys = v.RightJoinKeys
if v.InnerChildIdx == 1 && !v.UseOuterToBuild || v.InnerChildIdx == 0 && v.UseOuterToBuild {
	e.buildSideExec, e.probeSideExec = e.probeSideExec, e.buildSideExec
	e.buildKeys, e.probeKeys = e.probeKeys, e.buildKeys
}
if v.UseOuterToBuild {
	e.buildSideEstCount = v.Children()[1-v.InnerChildIdx].StatsCount()
}

if v.InnerChildIdx == 0 {
	if len(v.LeftConditions) > 0 {
		b.err = errors.Annotate(ErrBuildExecutor, "join's inner condition should be empty")
		return nil
	}
	if defaultValues == nil {
		defaultValues = make([]types.Datum, leftExec.Schema().Len())
	}
} else {
	if len(v.RightConditions) > 0 {
		b.err = errors.Annotate(ErrBuildExecutor, "join's inner condition should be empty")
		return nil
	}
	if defaultValues == nil {
		defaultValues = make([]types.Datum, rightExec.Schema().Len())
	}
}
```

Reply:
Although this can reduce the code length, as well as reduce the code readability.